app/api/views/brief_responses.py

```diff
@@ -122,3 +123,45 @@ def get_brief_response(brief_response_id):
                   .format(brief_response_id, current_user.supplier_code))
 
     return jsonify(brief_response.serialize())
+
+
+@api.route('/brief-response/<int:brief_id>/suppliers', methods=['GET'])
+@login_required
+@role_required('buyer')
+def get_suppliers_responded(brief_id):
+    """Get suppliers responded (role=buyer)
+    ---
+    tags:
+      - "Brief Response"
+    security:
+      - basicAuth: []
+    parameters:
+      - name: brief_response_id
+        in: path
+        type: number
+        required: true
+    definitions:
+      BriefResponse:
+        type: object
+        properties:
+          id:
+            type: number
+          data:
+            type: object
+          brief_id:
+            type: number
+          supplier_code:
+            type: number
+    responses:
+      200:
+        description: Suppliers that have responded to brief_id
+        schema:
+          id: BriefResponse
+      404:
+        description: brief_response_id not found
+    """
+
+    suppliers = brief_responses_service.get_suppliers_responded(brief_id)
```

Comment:
I don't think we're protecting against one buyer accessing the responding sellers of another buyer here, i.e. we should be validating ownership of the brief id in this endpoint.

Reply:
E.g.

```
brief = briefs.get(brief_id)
if not brief:
    not_found('brief {} not found'.format(brief_id))

brief_user_ids = [user.id for user in brief.users]
if not (
    hasattr(current_user, 'role') and
    (current_user.role == 'buyer' and current_user.id in brief_user_ids)
):
    forbidden('Unauthorised to view responding sellers')
```

Reply:
updated